File: app/src/main/java/no/nav/tilbakemeldingsmottak/config/WebConfig.java
package no.nav.tilbakemeldingsmottak.config;

import lombok.RequiredArgsConstructor;
import no.nav.tilbakemeldingsmottak.interceptors.MDCPopulationInterceptor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
@RequiredArgsConstructor
public class WebConfig implements WebMvcConfigurer {

    @Override
    public void addCorsMappings(CorsRegistry registry) {
        registry.addMapping("/rest/**")
                .allowedOrigins("http://localhost:3000", "http://localhost:2222")
                .allowedMethods("GET", "POST", "PUT", "DELETE", "OPTIONS", "HEAD", "PATCH")
                .allowCredentials(true)
                .maxAge(3600);
    }

	@Override
	public void addInterceptors(InterceptorRegistry registry) {
		registry.addInterceptor(new MDCPopulationInterceptor())
				.addPathPatterns("/rest/**");
    }
}

File: app/src/main/kotlin/no/nav/tilbakemeldingsmottak/config/WebConfig.kt
package no.nav.tilbakemeldingsmottak.config

import no.nav.tilbakemeldingsmottak.interceptors.MDCPopulationInterceptor
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebConfig : WebMvcConfigurer {
    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(MDCPopulationInterceptor())
            .addPathPatterns("/rest/**")
    }
}
